### ISSUE TITLE
Logistic Regression

### ISSUE THREAD

**Issue by ramyasri214** [ramyasri214/VERSION-CONTROL-WITH-GIT#1]
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
# Load dataset
data = pd.read_csv(r'C:\Users\PC\Desktop\ML Project\actual_abroad_data.csv')
# Preprocess: Create target variable based on FEES
data['Target'] = data['FEES'].apply(lambda x: "High Cost" if x > 400000 else "Low
Cost")
# Feature Encoding
data['COUNTRY'] = data['COUNTRY'].astype('category').cat.codes
data['COURSE TYPE'] = data['COURSE TYPE'].astype('category').cat.codes
data['COURSE (SPECIALIZATION)'] = data['COURSE
(SPECIALIZATION)'].astype('category').cat.codes
# Split data into features (X) and target (y)
X = data.drop(columns=['Target', 'FEES'])
y = data['Target']
# Train-Test Split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
# Logistic Regression Model
model = LogisticRegression()
model.fit(X_train, y_train)
y_pred = model.predict(X_test)
# Calculate metrics
accuracy = accuracy_score(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)
report = classification_report(y_test, y_pred, zero_division=0)
# Display metrics
print(f"Logistic Regression Model:\nAccuracy: {accuracy:.4f}")
print("Confusion Matrix:")
print(cm)
print(report)
# Visualize Confusion Matrix
plt.figure(figsize=(8, 6))
sns.heatmap(cm, annot=True, fmt='d', cmap='Greens', xticklabels=model.classes_,
yticklabels=model.classes_)
plt.title('Confusion Matrix for Logistic Regression Model')
plt.xlabel('Predicted Labels')
plt.ylabel('True Labels')
plt.show()

print("\nClassification Report:")
